### ISSUE TITLE
Unusual increase in active connections

### ISSUE THREAD

**Issue by adityaprakash-bobby** [GoogleCloudPlatform/cloudsql-proxy#580]
## Question
Hello, this query is related to #147, stating the same we observed a sudden increase in active database connections to around 252 and these connections remain as it is until we restart the container running the `cloud_sql_proxy`. After restarting the proxy container it runs with normal number of active connections (maximum that we have had around 12). We have been observing this behaviour quite recently which could have been occurring earlier too.

We are using a `ubuntu` image with `cloud_sql_proxy` installed. Which is then accessed by the application container to connect to the cloud SQL database (`Postgres`). Application uses `django` framework and all queries run through ORM.

Here's a snapshot of the same:
![Screenshot from 2020-12-24 20-07-11](https://user-images.githubusercontent.com/23385459/103096987-c3f3b980-462b-11eb-8321-8cf9bb7e78ab.png)

- `./cloud_sql_proxy -version`: 1.18.0

We observed this behaviour that it always goes to some number near about 250 active connections, drops to around 240 and remains in that range. And on restart we run with max a range of 8 active connections for a period until this happens. 

The high number of active connections does not affect the application but it is not the normal number of connections to have probably. What can we do to improve this issue? Will using a newer version of the proxy help? Or will using the pre-built docker images for cloud sql proxy helpful for this? Any input will be helpful.

Thanks!

**Comment by Carrotman42** [comment]
What application is connecting to the proxy? If we can reproduce on our own
it's much more likely to get solved, so if you can provide a minimal
reproducer that we can run on our own it would be very helpful to track
down the issue.

The proxy client *should* detect when the application closes it's
connection and then should close the connection to the remote database, but
I would be a bad engineer to assume that a bug is impossible :)

**Comment by adityaprakash-bobby** [comment]
Hi Kevin, seems we found out the issue. Basically it's Django, concurrency and mod_wsgi. It seems that they really don't play well together. It's something specific to the ORM and we were doing a blunder trying to make queries concurrently. A big lesson learnt. 

Going down the container logs, observing the connection pattern and issuing a similar concurrent script helped reproduce the issue, concluding that concurrency was the culprit here.

Apologies for the inconvenience. Thanks for taking out time to look out for this. Closing the issue here.
